packages/runtime/runtime-definitions/src/protocol.ts

```diff
@@ -126,3 +126,24 @@ export interface IRuntimeMessageCollection {
 	 */
 	readonly messagesContent: readonly IRuntimeMessagesContent[];
 }
+
+/**
+ * Outgoing {@link IFluidDataStoreChannel} message structures.
+ * @internal
+ *
+ * @privateRemarks
+ * Future use opportunity:
```

Comment:
Sounds nice!

Reply:
FYI @anthony-murphy

Reply:
Ok I'm talking out of both sides of my mouth - in the other comment I talked down adding these generics, didn't immediately connect the dots to this comment til now. I totally get the structure here and it's obviously appealing in one sense, but I'm concerned about how it will feel in practice.  Maybe too many generics?  Other downsides I'm not thinking of?